src/lightning/data/streaming/reader.py

```diff
@@ -24,6 +25,8 @@
 from lightning.data.streaming.sampler import ChunkedIndex
 from lightning.data.streaming.serializers import _SERIALIZERS, Serializer
 
+warnings.filterwarnings("ignore", message=".*The given buffer is not writable.*")
```

Comment:
You should do this in a function, or else it will get ignored on lightning.data import. Even for other libraries or user code

Reply:
done !